### PR TITLE
Ensured Output system for One Way frequencies worked correctly

### PR DESCRIPTION
**Pull request by derekagorhom**
fixes partly #8094 
@rdstern , @N-thony 
I was able to fix the output system for the `Table` and `graph` options of the dialogue and also i fixed the issue of the `stem and leaf` not producing any results

### CODE REVIEW

**Review by rdstern — CHANGES_REQUESTED** [review]
@derekagorhom that is much better now.  Could you change the code generated for the Table option, so it separates getting the variables from the sj function.  It now seems ok for the graph and I am happy we have the stem and leaf plot again.

**Comment by derekagorhom** [comment]
@rdstern 
i did the code for the output to display

`columns <- data_book$get_columns_from_data(data_name="survey", col_names=c("village","field"), remove_labels=TRUE, drop_unused_filter_levels=TRUE)`

`one_way_freq <- as.data.frame(x=sjmisc::frq(x=columns, show.strings=TRUE, show.na=TRUE, grp.strings=NULL))
data_book$import_data(data_tables=list(one_way_freq=one_way_freq))`

`data_book$get_data_frame(data_name="one_way_freq")`

`rm(list=c("one_way_freq", "columns"))` 

on the script widow.

**Comment by rdstern** [comment]
@derekagorhom thanks, this now seems to work fine.
However I am afraid I have stumbled on another small problem.  The earleir version permitted text or html output, with the html going to a browser.  Now we have html in the output window, but seem to have lost that option.  In the sjmisc::frq command it should come from adding out="viewer" to the function, and then the html option to the data_book_add_object function.  
This adds a third option to the dialogue.  @Patowhiz alternatively we now just have html as the output format in the output window?  What do you think?  I quite like the options of text or html, but don't feel strongly?

**Comment by derekagorhom** [comment]
@rdstern i managed to add the maximise to the output for the dataframe button.

File: instat/dlgOneWayFrequencies.vb [hunk 1]
```diff
@@ -22,6 +22,7 @@ Public Class dlgOneWayFrequencies
     Private clsSjMiscFrq As New RFunction
     Private clsSjPlot As New RFunction
     Private clsPlotGrid As New RFunction
+    Private clsGetColumnsFunction As New RFunction
```
Comment (N-thony) [inline instat/dlgOneWayFrequencies.vb:25]: Can you move this on before line 330 i.e Dim clsGetColumnsFunction As New RFunction?

File: instat/dlgOneWayFrequencies.vb [hunk 2]
```diff
@@ -164,6 +165,7 @@ Public Class dlgOneWayFrequencies
         clsAsDataFrame = New RFunction
         clsStemAndLeafFunction = New RFunction
         clsDummyFunction = New RFunction
+        clsGetColumnsFunction = New RFunction
```
Comment (N-thony) [inline instat/dlgOneWayFrequencies.vb:168]: then you remove this

File: instat/dlgOneWayFrequencies.vb [hunk 3]
```diff
@@ -216,7 +221,7 @@ Public Class dlgOneWayFrequencies
         ucrSaveGraph.AddAdditionalRCode(clsAsDataFrame)
 
         ucrReceiverWeights.SetRCode(clsSjMiscFrq, bReset)
-        ucrReceiverOneWayFreq.SetRCode(clsSjMiscFrq, bReset)
+        'ucrReceiverOneWayFreq.SetRCode(clsSjMiscFrq, bReset)
```
Comment (N-thony) [inline instat/dlgOneWayFrequencies.vb:224]: remove?

File: instat/dlgOneWayFrequencies.vb [hunk 4]
```diff
@@ -183,6 +185,9 @@ Public Class dlgOneWayFrequencies
 
         clsSjMiscFrq.SetPackageName("sjmisc")
         clsSjMiscFrq.SetRCommand("frq")
+        clsSjMiscFrq.AddParameter("show.strings", "TRUE")
+        clsSjMiscFrq.AddParameter("show.na", "TRUE")
+        clsSjMiscFrq.AddParameter("grp.strings", "NULL")
```
Comment (N-thony) [inline instat/dlgOneWayFrequencies.vb:190]: Can you add the positions?

**Review by N-thony — CHANGES_REQUESTED** [review]

**Comment by derekagorhom** [comment]
@N-thony I have made the changes requested.

**Review by rdstern — CHANGES_REQUESTED** [review]
@derekagorhom perhaps I didn't make myself clear last time.  Here is the dialogue in Version 0.7.8

image

You see that html output in the browser was then possible.  So now we no longer have html - and just when this is now an option in the output window.

You could add it, so the first option is txt, the second is the html - this time in the output window.  (That was supposed to be the main change with the new output window.)

I don't have a problem with the addition of the new option as dataframe, so that can stay if you wish.

**Comment by Patowhiz** [comment]
@derekagorhom I think we discussed with @anastasia-mbithe and @Vitalis95 about similar options. Check with them.

**Comment by derekagorhom** [comment]
@rdstern I was able to implement the HTML option to this dialogue, but the problem is that it is not able to produce any results in the output window so i made the result display in the browser. 
I have had conversations with both Anastasia and Vitalis and we still don't know why it does not produce the HTML result in the output.

**Comment by Patowhiz** [comment]
@derekagorhom I'll now work on this.

**Comment by rdstern** [comment]
@Patowhiz will you still be doing this?  I am keen to "finish" all these options for the Describe > One Variable and 2/3 Variable as soon as possible.

**Comment by N-thony** [comment]
> @Patowhiz will you still be doing this? I am keen to "finish" all these options for the Describe > One Variable and 2/3 Variable as soon as possible.

@derekagorhom is this fixed/replaced by another PR already? If so, can you close this PR?

**Pull request closed without merging**